src/api/general-services/work-submit/index.js

```diff
@@ -86,16 +88,31 @@ class WorkSubmitService {
   async createWorker() {
     if (config.clusterEnv === 'development' || config.clusterEnv === 'test') {
       logger.log('Not creating a worker because we are running locally...');
+
       return;
     }
 
     await createWorkerResources(this);
   }
 
   async submitWork() {
+    AWSXRay.getSegment().addAnnotation('result', 'success-worker');
+    AWSXRay.getSegment().close();
```

Comment:
Is this order correct? Seems counter-intuitive to close the segment and later close subsegments...

Reply:
It is correct, it is a weird behavior but it is precisely what we need -- the original segment is closed because we responded to the user, but the subsegments are not done until the promises resolve. You can do this, even though it is weird, and XRay supports it.